src/orchestrator/chat_completions_detection.rs

```diff
@@ -82,6 +85,25 @@ impl From<&ChatCompletionsRequest> for Vec<ChatMessageInternal> {
     }
 }
 
+// Get Vec<ChatMessageInternal> from ChatCompletion
+impl From<&Box<ChatCompletion>> for Vec<ChatMessageInternal> {
+    fn from(value: &Box<ChatCompletion>) -> Self {
+        value
+            .choices
+            .iter()
+            .enumerate()
+            .map(|(index, choice)| ChatMessageInternal {
+                message_index: index,
```

Comment:
Not sure if choices are guaranteed to be in order, so should we use `choice.index` directly here instead of going through the `enumerate` path?